src/webapi/blob.rs

```diff
@@ -31,6 +34,46 @@ pub trait IBlob: ReferenceType {
             Some( mime )
         }
     }
+
+    /// Create a new `Blob` object containing the data in the specified range of bytes of the
+    /// source `Blob`.
+    /// 
+    /// See also [slice_with_content_type](IBlob::slice_with_content_type).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice)
+    // https://w3c.github.io/FileAPI/#ref-for-dfn-slice
+    fn slice< T >( &self, range: T ) -> Blob
+        where T: RangeBounds<u64>
+    {
+        self._slice(range, None)
+    }
+
+    /// [slice](IBlob::slice) `Blob` with the provided `content_type`.
+    fn slice_with_content_type< T >( &self, range: T, content_type: &str ) -> Blob
+        where T: RangeBounds<u64>
+    {
+        self._slice(range, Some(content_type))
+    }
+
+    #[doc(hidden)]
+    fn _slice< T >( &self, range: T, content_type: Option<&str> ) -> Blob
```

Comment:
It's actually not necessary to make this a trait method. You can just make it into a freestanding function, e.g. something like this:

```rust
fn slice_blob< T, R >( blob: &T, range: R, content_type: Option< &str > ) -> Blob
    where T: IBlob, R: RangeBounds< u64 >
{
...
}
```

and then just call it from the actual trait methods.

Reply:
I notice that you are using some formatting conventions in codes which are different from those enforced by the default rustfmt. Do you have a formatter / configurations so that I can adhere easily, if necessary?